### PR TITLE
com1DFAPy run script and tests

### PR DESCRIPTION
**Pull request by awirb**
* simplify com1DFAPy run script and make callable 
* introduce ana1Tests
* add dam break test 
* simplify run script for simi sol and move to ana1Tests
* add data/avaDamBreak and data/avaSimiliaritySol

### CODE REVIEW

**Review by fso42 — CHANGES_REQUESTED** [review]
I did not go through all files yet, because I think there's a big change needed before we move this to master:

Both damBreak and simiSol in ana1Test mix calculations and plotting within the same functions. I think this should only be done in run* scripts (or in full on development modules like com1DFApy). 

SimiSol even has a setup at the top which is not behind a function -> therefore run every time one imports it. 

I think the functions in damBreak/simiSol should contain:
- separate functions for generating the variables / solutions (and smaller helper functions)
- if necessary special plotting routines (*if they don't fit into out3Plot*)

What I think should be possible:
- Generate the solutions, get back the variables for them 
- Then compare this to ANY model one wants to  run
- Either do this by some comparison or plotting (if wanted) 

These test should be independent of the model one wants to investigate.

If there is an i.e. compareSimiToCom1DFAPy (or similar) in the damBreak.py file, that's fine, as long as it uses functions in that can be called independently. (However I feel this should be done in a run*.py script)

A few minor things I saw:
- namev should be namefv (to follow namepfv)
- there's still a myDEM_IP -> should be the one without `my`

File: avaframe/ana1Tests/simiSol.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+# imports
+import numpy as np
+from scipy.integrate import ode
+import math
+import os
+
+# local imports
+from avaframe.in3Utils import cfgUtils
+import avaframe.com1DFAPy.com1DFA as com1DFA
+import avaframe.ana1Tests.simiSol as simiSol
```
Comment (matthiasto) [inline avaframe/ana1Tests/simiSol.py:18]: is this import still required?

**Review by matthiasto — CHANGES_REQUESTED** [review]

File: avaframe/ana1Tests/simiSol.py (hunk 1 above)
Comment (awirb) [reply avaframe/ana1Tests/simiSol.py:21]: I removed unused imports

File: avaframe/out3Plot/plotUtilsCfg.ini [hunk 2]
```diff
@@ -51,3 +51,4 @@ namepfd = peak flow depth
 namepfv = peak flow velocity
 namerelTh = release thickness
 namefd = flow depth
+namev = flow velocity
```
Comment (fso42) [inline avaframe/out3Plot/plotUtilsCfg.ini:54]:
```suggestion
namefv = flow velocity
```

File: avaframe/runSimilaritySol.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+# +++++++++POSTPROCESS++++++++++++++++++++++++
+# -------------------------------
+# Analyse resutls
+# partRef = Particles[0]
+# Z0 = partRef['z'][0]
+# gravAcc = cfg['GENERAL'].getfloat('gravAcc')
+# # Dimentioning parameters
+# U = np.sqrt(gravAcc*L_x)
+# V = np.sqrt(gravAcc*L_y)
+# mu = cfgGen.getfloat('mu')
```
Comment (fso42) [inline avaframe/runSimilaritySol.py:72]: remove this?

**Review by fso42 — CHANGES_REQUESTED** [review]
See the inline comment, plus

- theres myDEM_IP_Topo.asc showing up -> move to DEM_IP_Topo...

File: avaframe/com1DFAPy/runCom1DFA.py [hunk 4]
```diff
@@ -1,194 +1,125 @@
...
-# ax.plot(T, U, 'k', linestyle='-', linewidth=2)
-# ax.plot(T, Z, 'b', linestyle='-')
-# ax.plot(T, np.sqrt(2 * gravAcc * ((Z0-Z) - mu * S)), 'r', linestyle='-', linewidth=1)
-# # ax.plot(T, np.sqrt(2 * gravAcc * ((Z0-Z))), 'r', linestyle='-', linewidth=1)
-# plt.show()
+
+def setDEMorigin(demOri):
+    """ set origin of DEM to 0,0 """
+
+    dem = copy.deepcopy(demOri)
```
[the remark this answers is not in the document]
Comment (matthiasto) [reply avaframe/com1DFAPy/runCom1DFA.py:23]: I want to compute the simulation with (0, 0) as the lower left center.  I made a copy of the dem with xllc and yllc set to zero and give this to the computation module. This way the computation is done using the new dem but still using operations such as iLx = (x - xllc) / csz, which are general functions.

File: avaframe/ana1Tests/damBreak.py [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+            cond1 = ((m0*t[m]) / 2.0 - cL) * t[m]
+            cond2 = (2.0 *cL + ((m0*t[m]) / 2.0)) * t[m]
+            if x[k] <= cond1:
+                u[k,m] = m0 * t[m]
+            elif cond1 < x[k] <= cond2:
+                u[k,m] = (2./3.) * (cL + (x[k] / t[m]) + m0 * t[m])
+
+    #-----------------------------Plot results --------------
+    # Reproduce figure 6, case 1.2.1 - Test 2
+    plotResults(x, h, u, dtStep, cfg)
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana1Tests/damBreak.py:229]: plots are also saved, but in the plot functions there is a flag showplot

File: avaframe/ana1Tests/simiSol.py [hunk 6]
```diff
@@ -0,0 +1,521 @@
...
+import avaframe.in2Trans.ascUtils as IOf
+import avaframe.out3Plot.plotUtils as pU
+
+
+# create local logger
+# change log level in calling module to DEBUG to see log messages
+log = logging.getLogger(__name__)
+
+
+def define_earth_press_coeff(phi, delta):
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana1Tests/simiSol.py:31]: done

File: avaframe/ana1Tests/simiSol.py [hunk 7]
```diff
@@ -0,0 +1,521 @@
...
+    planeinclinationAngleDeg = cfgSimi.getfloat('planeinclinationAngle')
+    x = np.linspace(0, ncols-1, ncols)*csz+xllc
+    y = np.linspace(0, nrows-1, nrows)*csz+yllc
+    X, Y = np.meshgrid(x, y)
+    cos = math.cos(math.pi*planeinclinationAngleDeg/180)
+    sin = math.sin(math.pi*planeinclinationAngleDeg/180)
+    X1 = X/cos
+    Y1 = Y
+    r = np.sqrt((X*X)/(cos*cos)+(Y*Y))
+    H0 = Hini
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana1Tests/simiSol.py:346]: done

File: avaframe/ana1Tests/simiSol.py [hunk 8]
```diff
@@ -0,0 +1,521 @@
...
+        plt.pause(1)
+
+    fig, ax, cmap, lev = com1DFA.plotContours(
+        fig, ax, part, demOri, field['FD'], pU.cmapDepth, 'm', last=True)
+    CS = ax.contour(X, Y, hSimi, levels=lev, origin='lower', cmap=cmap,
+                    linewidths=2, linestyles='dashed')
+    ax.clabel(CS, inline=1, fontsize=8)
+
+
+def prepareParticlesFields(Fields, Particles, ind_t, relDict, simiDict, axis):
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana1Tests/simiSol.py:390]: Done

File: avaframe/com1DFAPy/com1DFA.py [hunk 9]
```diff
@@ -1079,3 +1099,91 @@ def removePart(particles, mask, nRemove):
     particles['partInCell'] = particles['partInCell'][mask]
 
     return particles
+
+def exportFields(cfgGen, Tsave, Fields, relFile, demOri, outDir):
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py:1103]: This is a general function to export fields of com1DFAPy, so far there are only null sims, but I changed the 0.155 to use mu from the ini file

File: avaframe/runSimilaritySol.py (hunk 3 above)
Comment (awirb) [reply avaframe/runSimilaritySol.py:72]: done

File: avaframe/out3Plot/plotUtilsCfg.ini (hunk 2 above)
Comment (awirb) [reply avaframe/out3Plot/plotUtilsCfg.ini:54]: I made an issue out of this, as this has to be then renamed everywhere in com1DFAPy

File: avaframe/com1DFAPy/com1DFA.py [hunk 10]
```diff
@@ -1079,3 +1099,91 @@ def removePart(particles, mask, nRemove):
...
+
+    cfgGen = cfg['GENERAL']
+    partRef = Particles[0]
+    Z0 = partRef['z'][0]
+    rho = cfgGen.getfloat('rho')
+    gravAcc = cfgGen.getfloat('gravAcc')
+    mu = cfgGen.getfloat('mu')
+    repeat = cfg['DEVFLAGS'].getboolean('debugPlot')
+    while repeat:
+        fig, ax = plt.subplots(figsize=(pU.figW, pU.figH))
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFAPy/com1DFA.py:1160]: Done

File: avaframe/ana1Tests/damBreak.py [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+
+    fig = plt.figure(figsize=(pU.figW, pU.figH))
+    plt.title('Dry-Bed, $\delta=21, \phi$=22')
+    plt.plot(x, u[:,0], 'k--', label='t_init')
+    plt.plot(x, u[:,dtInd], label='t = %.1fs' % dtStep)
+    plt.xlabel('x-coordinate [m]')
+    plt.ylabel('velocity [ms-1]')
+    plt.legend()
+
+    fig.savefig(os.path.join(outDir, 'damBreakVelocity.%s' % (pU.outputFormat)))
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana1Tests/damBreak.py:62]: Done

File: avaframe/ana1Tests/damBreak.py [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+    ax.plot(x, y, 'grey', linestyle='--')
+    ax.plot(x, dataIniV[nx_loc, :], 'k--', label='init')
+    ax.plot(x, dataAnaV[nx_loc, :], 'b', label='com1DFAPy')
+    ax.plot(xR, uR[:,tR], 'r-', label='analyt')
+    ax.set_xlabel('Along track [ncols]')
+    ax.set_ylabel('Flow velocity [ms-1]')
+    plt.legend()
+    ax.set_title('Flow velocity at time step %.02f s' % (dtAnalysis))
+
+    fig2.savefig(os.path.join(outDir, 'CompareDamBreakVel.%s' % (pU.outputFormat)))
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana1Tests/damBreak.py:174]: done

File: avaframe/ana1Tests/simiSol.py [hunk 13]
```diff
@@ -0,0 +1,521 @@
...
+            fig, ax, part, demOri, field['FD'], pU.cmapDepth, 'm')
+        CS = ax.contour(X, Y, hSimi, levels=lev, origin='lower', cmap=cmap,
+                        linewidths=2, linestyles='dashed')
+        plt.pause(1)
+
+    fig, ax, cmap, lev = com1DFA.plotContours(
+        fig, ax, part, demOri, field['FD'], pU.cmapDepth, 'm', last=True)
+    CS = ax.contour(X, Y, hSimi, levels=lev, origin='lower', cmap=cmap,
+                    linewidths=2, linestyles='dashed')
+    ax.clabel(CS, inline=1, fontsize=8)
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana1Tests/simiSol.py:386]: done

File: avaframe/ana1Tests/simiSol.py [hunk 14]
```diff
@@ -0,0 +1,521 @@
...
+    uySimi = np.where(hSimi <= 0, 0, uySimi)
+    vSimi = np.sqrt(uxSimi*uxSimi + uySimi*uySimi)
+    xCenter = simiSol.xc(solSimi, X1, Y1, ind_time, L_x)*cos
+
+    simiDict = {'hSimi': hSimi, 'vSimi': vSimi, 'xCenter': xCenter}
+
+    return simiDict
+
+
+def plotProfilesSimiSol(ind_time, relDict, com1DFAPySol, simiDict, solSimi, axis):
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/ana1Tests/simiSol.py:457]: It is already but I renamed it to make it clearer

**Review by fso42 — APPROVED** [review]

File: avaframe/com1DFAPy/com1DFACfg.ini [hunk 15]
```diff
@@ -3,6 +3,10 @@
 # This file is part of Avaframe.
 
 
+[DEVFLAGS]
+debugPlot = True
+
+
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFAPy/com1DFACfg.ini:9]: made an issue out of this

**Review by matthiasto — APPROVED** [review]

**Pull request merged**